### PR TITLE
Fix: MethodChannel is called for platforms that don't have some implementations such as endNativeFrames

### LINKED ISSUE #1176: `MethodChannel` is called for platforms that dont have some implementations such as `endNativeFrames`
### Description

For example https://github.com/getsentry/sentry-dart/blob/d235a79dc30772bb861de3eec8222cf2300bbd23/flutter/lib/src/sentry_native_channel.dart#L39-L44
On the Web, `endNativeFrames` returns an empty string so it fails to cast to `<String, dynamic>`.
Since we swallow its exceptions, it's not a problem, but we should not call such methods in the first place, only for Android, iOS, and macOS for now.

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Closes #1176

## :green_heart: How did you test it?

Added unit tests.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: flutter/ios/Classes/SentryFlutterPluginApple.swift [hunk 1]
```diff
@@ -86,6 +86,11 @@ public class SentryFlutterPluginApple: NSObject, FlutterPlugin {
             let value = arguments?["value"] as? Any
             setContexts(key: key, value: value, result: result)
 
+        case "removeContexts":
```
Comment (denrase) [inline flutter/ios/Classes/SentryFlutterPluginApple.swift:89]: @marandaneto Call to this code was missing, but i think we don't use it anyway, as we use "loadContexts" instead to merge scope.

Comment (marandaneto) [reply flutter/ios/Classes/SentryFlutterPluginApple.swift:89]: Good catch, maybe this should make into the main branch together with the fix of the scope sync, I guess this PR wont have any conflicts if you change the target branch, right?

**Review by marandaneto — APPROVED** [review]
Approving it, but maybe we can cut this fix together with the scope sync in the main branch.

**Comment by denrase** [comment]
Closing in favour of #1196

**Pull request closed without merging**